### PR TITLE
flexible PSTH for foraging task

### PR DESCRIPTION
**Pull request by hanhou**
1. added a new table `psth_foraging.AlignType` to define arbitrary temporal alignments
2. add `plot.unit_psth.plot_unit_psth_choice_outcome`

### CODE REVIEW

**Review by ttngu207 — COMMENTED** [review]
Overall I really like your approach of the `AlignType` table, in combination with the `compute_unit_psth_and_raster()` function, allowing for very flexible PSTH computation.
The logic is complex, for good reason, I'd suggest to add a bit more documentation in the docstring to describe, on high level, the approach taken by using this `AlignType` table and `compute_unit_psth_and_raster()` function.

File: pipeline/psth_foraging.py [hunk 1]
```diff
@@ -349,3 +416,103 @@ def get_plotting_data(cls, unit_key, condition_key):
...
+    definition = """
+    idx:                 smallint
+    -> experiment.TrialEventType
+    ---
+    align_type_name:     varchar(128)     # user-friendly name of alignment type
+    trial_offset=0:      smallint         # e.g., offset = 1 means the psth will be aligned to the event of the *next* trial
+    time_offset=0:       Decimal(10, 5)   # will be added to the event time for manual correction (e.g., bitcodestart to actual zaberready)  
+    psth_win:            tinyblob   
+    xlim:                tinyblob
+    """
```
[the remark this answers is not in the document]
Comment (hanhou) [reply pipeline/psth_foraging.py:435]: Thanks! I think using `align_type_name` as a primary key indeed makes more sense. 
I'm also happy to learn another (undocumented?) usage `unique index` !

[the remark this answers is not in the document]
Comment (hanhou) [reply pipeline/psth_foraging.py:435]: Actually `-> experiment.TrialEventType` could be moved to a secondary key, right?

**Comment by hanhou** [comment]
Thanks! Yes, it's a bit complex. I've added some more comments to the docstring of `class AlignType` and `unit_psth.plot_unit_psth_choice_outcome`

**Review by ttngu207 — APPROVED** [review]

**Pull request merged**